src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,264 @@
+import duke.event.Event;
+import duke.event.ToDo;
+import duke.task.Task;
+import netscape.javascript.JSObject;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.JSONValue;
+
+public class Duke {
+    private static final String MARK_CMD = "mark";
+    private static final String UNMARK_CMD = "unmark";
+    private static final String LIST_CMD = "list";
+    private static final String ADD_CMD = "add";
+    private static final String DELETE_CMD = "delete";
+    private static final String OUT_DIR = "output";    // Honestly create a config file or smth
+    private static final Boolean isDebugMode = false;
+    private static final String SAVE_PATH = String.format("%s/%s", OUT_DIR, "SAVE.json");
+
+    private static List<Task> toDoList = new ArrayList<Task>();
+
+    private static void serializeTaskList() {
+        JSONArray toStore = new JSONArray();
+        for (Task task : toDoList) {
+            toStore.add(task.serialize());
+        }
+        String storeStr = toStore.toString();
+        System.out.println(toStore);
+        Path dir = Paths.get(OUT_DIR);
+//        File f = new File(OUT_DIR);
+        if (!Files.exists(dir)) {   //createTempDirectory
+            try {
+                Files.createDirectory(Path.of(OUT_DIR));
+            } catch (IOException e) {
+                System.err.println("Failed to create directory!" + e.getMessage());
+            }
+        }
+//        String filePath = String.format("%s/%s", OUT_DIR, "SAVE.json");
+        try {
+            FileWriter fw = new FileWriter(SAVE_PATH);
+            fw.write(storeStr);
+            fw.close();
+        } catch (IOException e) {
+            System.err.println("Failed to write to save file!" + e.getMessage());
+        }
+    }
+
+    private static void loadTaskList() {
+        String saveStr = "";
+        try {
+            FileReader fr = new FileReader(SAVE_PATH);
+            int i;
+            while ((i=fr.read()) != -1){
+                saveStr += ((char)i);
+            }
+            System.out.println(saveStr);
+            fr.close();
+        } catch (IOException e){
+            System.err.println("Failed to open save file!" + e.getMessage());
+            return;
+        }
+        // ATTEMPTING PARSE
+
+        JSONArray ja = (JSONArray) JSONValue.parse(saveStr);
+//        Object[] ja = ((JSONArray) JSONValue.parse(saveStr)).toArray();
+//        for (int i = 0; i < ja.length; i++){
+//            Object curItem = ja[i];
+//            toDoList.add(new Task())
+//        }
+        for(Object o: ja){
+            if ( o instanceof JSONObject) {
+                JSONObject jo = (JSONObject)o;
```

Comment:
You could use a proper name for the json object - see [how to name well](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#guideline-name-well).